lib/pbench/test/unit/server/database/test_server_config_db.py

```diff
@@ -183,33 +183,44 @@ def rollback(self):
         """
         self.rolledback += 1
 
+        # Clear the proxy state by removing any new objects that weren't yet
+        # committed, and "rolling back" any proxy values that were updated
+        # from the committed values.
+        self.added = []
+        for k in self.committed:
+            self.known[k].value = self.committed[k].value
+
 
 class TestServerConfig:
     session = None
 
-    def check_session(self, added=[], queries=0, committed=[], rolledback=0):
+    def check_session(self, queries=0, committed: List[FakeRow] = [], rolledback=0):
         """
         Encapsulate the common checks we want to make after running test cases.
 
         Args:
-            added: A list of object proxies we expect to have been added (but
-                not yet committed)
             queries: A count of queries we expect to have been made
-            committed: A set of objects we expect to have been committed
+            committed: A list of FakeRow objects we expect to have been
+                committed
             rolledback: True if we expect rollback to have been called
         """
         session = self.session
         assert session
+
+        # Check the number of queries we've created
         assert len(session.queries) == queries
-        assert len(session.added) == len(added)
-        xlated = [
-            a
-            for a in session.added
-            for k in added
-            if a.key == k.key and a.value == k.value
-        ]
-        assert session.added == xlated
+
+        # 'added' is an internal "dirty" list between 'add' and 'commit' or
+        # 'rollback'. We test that 'commit' moves elements to the committed
+        # state and 'rollback' clears the list. We don't ever expect to see
+        # anything on this list.
```

Comment:
>  We don't ever expect to see anything on this list.

Only, if we are careful about when we call `check_session()`, right?  (That is, this assertion places a constraint on where/when this function can be invoked, right?...we're assuming that it will only be used after a commit or failure/rollback.)

Reply:
OK, nevermind:  at the level where this function is being called, we never have the opportunity to peek into the middle of a transaction, so, yeah, at the points where this function might be called, there should never be anything on the list.  (My bad.)